### PR TITLE
debug: gdbstub:  Move to DTS for uart device

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Move from using Kconfig GDBSTUB_SERIAL_BACKEND_NAME to a devicetree
chosen property ("zephyr,gdbstub-uart"). This is similar to a number
of other functions like "zephyr,shell-uart" or "zephyr,bt-uart".

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

File: samples/subsys/debug/boards/qemu_x86_64.overlay [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/ {
+        chosen {
+                zephyr,gdbstub-uart = &uart1;
+        };
+};
```
Comment (gmarull) [inline samples/subsys/debug/boards/qemu_x86_64.overlay:7]: please use tabs ;-)

**Review by gmarull — CHANGES_REQUESTED** [review]
update `doc/build/dts/api/api.rst` list as well

**Review by gmarull — DISMISSED** [review]

**Review by galak — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

File: doc/build/dts/api/api.rst [hunk 2]
```diff
@@ -410,6 +410,8 @@ device.
    * - zephyr,flash-controller
      - The node corresponding to the flash controller device for
        the ``zephyr,flash`` node
+   * - zephyr,gdbstub-uart
+     - Sets UART device used by gdbstub debug subsystem
```
Comment (mbolivar-nordic) [inline doc/build/dts/api/api.rst:414]:
```suggestion
     - Sets UART device used by the :ref:`gdbstub` subsystem
```

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
LGTM, but please add a link to the relevant docs page, to make it easier for users to find. See suggestion.

**Review by dcpleung — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by galak** [comment]
@mbolivar-nordic can you take another look.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**